### ISSUE TITLE
Google Cast does not work (icon missing)

### ISSUE THREAD

**Issue by RhavoX** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#3605]
**Checklist:**

* [ ] I have checked that there are no issues with similar or the same content

  **YOU SHOULD CHECK CLOSED ISSUES ASWELL**
* [ ] I have checked the FAQ (https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/wiki/FAQ) and the answer I am looking for is not there
* [ ] I have double checked and can reproduce the issue

**OS:**
Windows 10 1903

**GPMDP Version:**
4.6.1

**Issue Descriptions:**
There is no cast button and I'm unable to locate it in any menus  (I do have compatible speakers of course :) - Polk MagniFi MAX)

**Steps to Reproduce:**
Start the application

**Comment by jostrander** [comment]
Do you have other chromecast devices you can test with to make sure the cast functionality does work for you?

**Comment by gwilym** [comment]
While I'm not the original reporter I'm having the same issues on Windows 10 1903 and a Chromecast Ultra: no "cast" icon visible in the UI.

The Desktop Player running on my MacOS works. Chrome's own "cast" dialog works on both. The  Desktop Player on Windows does not.

Restarting the Chromecast Ultra does not fix the issue.

**Comment by Mustache84** [comment]
I am experiencing the same situation. :(

It used to work, now it doesn't. I'm unsure of what changed other than OS updates. 

Windows 10 1903, build 18362.356

**Comment by Mustache84** [comment]
Updates? comments? devs? I want to donate to this project but when almost 2 months go by without a response I kind of don't feel like this is under much development :O

**Comment by jostrander** [comment]
This is probably due to a firewall rule. Chrome somehow opens port 5353 for itself, but the method used by this repo  hasn't been updated in a few years. I would white list port 5353 UDP in your firewall for GPMDP and see if that helps.

**Comment by chrisreedio** [comment]
> I am experiencing the same situation. :(
> 
> It used to work, now it doesn't. I'm unsure of what changed other than OS updates.
> 
> Windows 10 1903, build 18362.356

I also have lost my cast button in my Desktop client. 
I have had it and used it regularly for over a year now with my Google Home, Google Home Mini, and Chromecast Audio. It also worked great with my Home Group.

I opened the port 5353 UDP in my firewall (inbound/outbound AND for all applications, just to be sure) and I still do not have the icon. 

Any more ideas that I can try? 

As with @Mustache84 I recently updated my Windows 10 as it was probably a year or more behind on updates. I figured as you said this created or removed some firewall rules which was blocking communications with the Google Home environment. Unfortunately that didn't solve my issue.

**Comment by jostrander** [comment]
You may have to install bonjour print services. It's what allows this app to search via mDNS on mac, and possibly windows, iirc. Just throwing it out there, that may be a thing of the past.

**Comment by thombien** [comment]
I have the same Issue I updated to the last unoficial release since i got windows 10 when launched  the app and I dont see any Cast for my Google Home device
[debug_info.zip](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/files/3959306/debug_info.zip)

**Comment by cmcinroy** [comment]
I am having this problem, too.
Ubuntu 18.04.3 LTS x86_64, kernel 4.15.0-72-generic.

I installed the build from the CircleCI 3751 job in order to get past the "stuck on loading" problem (#3660 ). Now there is no "cast" icon visible.

**Comment by NiborIS** [comment]
I am in the same position as @cmcinroy and for the same reason.

However, I am using Win 10 [Version 10.0.19041.1]

Thanks.

**Comment by Sleepybear** [comment]
Same, on latest Win10 build the Cast button has disappeared. 
It has also disappeared from the mobile app, and for some reason the Web player won't let me cast to my Google home mini, says unsupported (perhaps because no screen and chrome wants a screen?)

Please fix Casting from desktop app!

**Comment by ChildishGiant** [comment]
I also have no option on windows 10

**Comment by minohco** [comment]
Cast button shows up just fine in GPM with 4.7 on 1909 win10pro. YouTube Music interface does not have a cast button - any way to add a button or keyboard shortcut to bring up cast menu?

**Comment by e2zippo** [comment]
Same here, I don't see a cast icon, works fine in Chrome web browser.

**Comment by Sleepybear** [comment]
With newest build I have the cast button when using Google Play Music, but then I accidentally switched to youtube music and the cast button is not available there

**Comment by crzdmniac** [comment]
Just to add, I'm having the same issues as everyone here has mentioned. I've removed all the settings from appdata and still do not have the cast button. I do however have two "show mini player" buttons, one is where the cast button normally would be. Thanks for a great program.

**Comment by thomaszeroone** [comment]
I just wanted to chime in here. I'm on Win10. Cast button appears in both Chrome and the android app. I really want to support this desktop version but being able to use chromecast is a strict requirement for me!

**Comment by Sleepybear** [comment]
Cast button is now showing up and working for me for both GMP and YTM. Note that if no cast devices are found it won't show up

**Comment by veeeeeeeeeeee** [comment]
Having the same issue, installing bonjour does NOT fix it. I closed the player and just installed it for testing. The icon is missing from the player so I'm not sure why anyone thinks it will fix this issue?

This was working fine until last week, I can still broadcast directly from googleplaymusic via my browser, just not the standalone app.

The icon has totally disappeared, there are no settings that disable/enable this.
I uninstalled it the other day and reinstalled it with no luck.

I do not have HyperV or VMware installed as per previous suggestions mentioned.

Windows 10 Pro:OS Version: 10.0.18363 N/A Build 18363

**Comment by andrewalpha** [comment]
When I installed the program, I was able to connect to Chromecast. Then the button disappeared. Maybe had something to do with the device continuing to play when I close the program?
I did a system restore, then reinstalled the program. The button reappeared!
Then disappeared again, soon as I closed and reopened the program.

**Comment by e2zippo** [comment]
Sadly this is deal breaker for me, since I mostly cast my music. I really wish this could be solved but it doesn't look like it's getting any attention.

**Comment by Sleepybear** [comment]
My cast button appears and disappears on a whim. 

I'm thinking it has to do with how the window is rendered, which I am assuming uses a web viewer as the base. If my SoundBar is turned off, it doesn't seem to show up at all, so could have something to do with the discoverablility of the devices?

It would be nice to have the cast button always, and if no devices are found to show a message when clicked.

**Comment by e2zippo** [comment]
It should be handled like Chrome handles it. No matter what I do I can't get the cast icon to show, when it shows all the time in Chrome. Are people still getting updates through the app? I haven't seen that in a long time.

**Comment by Sleepybear** [comment]
Yes it should, but you will notice if you use the music.google.com web UI, there is also sometimes a Cast button there, but it then delegates it to the built in chrome one. I notice that the cast button in the UI disappears when the Chrome cast button doesn't have any "Supported devices"

**Comment by ryanrapini** [comment]
May not be the direct solution to your issues, but I'm leaving this here for people (like me) who found this thread via Google. Installing [Bonjour Print Services](https://download.cnet.com/Bonjour-Print-Services/3000-2088_4-77626130.html) fixed the issue for me completely.

**Comment by rileym94** [comment]
> 
> 
> With newest build I have the cast button when using Google Play Music, but then I accidentally switched to youtube music and the cast button is not available there

I don't believe YTM casting is supported yet. I was just digging around to see if there was any information on it. 

_Correct me if I'm wrong, folks._ ;)

**Comment by cmcinroy** [comment]
FYI, It appears there is the ability to cast from the Youtube Music service in the browser. Clicking the cast icon in the upper right opens the same window to select the target casting device as when one navigates the broswer menu to cast a tab.
